### PR TITLE
[BEAM-886] Migrate from OldDoFn to NewDoFn

### PR DESCRIPTION
**Pull request by sb2nov**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

R: @aaltay PTAL

### CODE REVIEW

**Comment by sb2nov** [comment]
R: @aaltay PTAL

File: sdks/python/apache_beam/examples/wordcount_debugging.py [hunk 1]
```diff
@@ -46,14 +46,15 @@
...
+from apache_beam import NewDoFn
 from apache_beam.io import ReadFromText
 from apache_beam.io import WriteToText
 from apache_beam.metrics import Metrics
 from apache_beam.utils.pipeline_options import PipelineOptions
 from apache_beam.utils.pipeline_options import SetupOptions
 
 
-class FilterTextFn(beam.DoFn):
+class FilterTextFn(NewDoFn):
```
Comment (aaltay) [inline sdks/python/apache_beam/examples/wordcount_debugging.py]: Why not `beam.NewDoFn` ?

**Comment by sb2nov** [comment]
Yes all the DoFns in beam should be migrated after this.

File: sdks/python/apache_beam/examples/wordcount_debugging.py (hunk 1 above)
Comment (sb2nov) [inline sdks/python/apache_beam/examples/wordcount_debugging.py]: Will change.

**Comment by aaltay** [comment]
LGTM

**Comment by sb2nov** [comment]
Tests passed so should be ready to merge

**Pull request closed without merging**